pkg/plugins/golang/v2/api.go

```diff
@@ -142,15 +138,30 @@ func (p *createAPISubcommand) Validate() error {
 		p.options.DoController = util.YesNo(reader)
 	}
 
+	// Create the resource from the options
+	p.resource = p.options.NewResource(p.config)
+
+	return cmdutil.Run(p)
+}
+
+func (p *createAPISubcommand) Validate() error {
+	if err := p.options.Validate(); err != nil {
+		return err
+	}
+
```

Comment:
Won't `Options.Validate()` be a subset of `Resource.Validate()`?
```suggestion
```

Reply:
`Options.Validate` only does the validation directly related to the flag parsing. The idea is that third-party pugins don't have to use the `Options` objects as those are just how go plugin flag options are parsed, but still benefit from the validation that was previously being done there and now is mvoed to `Resource.Validate()`.

Reply:
Sure the `Options.Validate()` method should still exist. I'm suggesting that in this particular method that calling `Options.Validate()` is redundant since `Resource.Validate()` is called afterwards, and `p.options` isn't used in `cmdutil.Run(p)`.

Reply:
`Options.Validate()` still does some validations like checking that the values for the flags are not empty. Those validations do not belong to the `Resource` itself.